charts/consul/templates/crd-meshgateways.yaml

```diff
@@ -56,6 +56,16 @@ spec:
                 description: GatewayClassName is the name of the GatewayClass used
                   by the MeshGateway
                 type: string
+              listeners:
+                items:
+                  properties:
+                    name:
+                      type: string
+                    port:
+                      format: int32
```

Comment:
Hmm... will there be issues without setting a min here? In the proto definition we have an unsigned integer, which openapi spec doesn't support. It seems like setting `-1` as a listener port at the CRD level would subsequently pass the spec, and I'm guessing depending on how the json unmarshaler handles things we'd either get some weird error unmarshaling the payload and just continually retry, or we'd get some sort of underflow and assign an invalid port number.

Thinking it'd make sense to pin a min/max of 0/65535 for this field.

Reply:
Working through setting the min, this PR needs to be merged first. https://github.com/hashicorp/consul/pull/20120

Reply:
https://github.com/hashicorp/consul-k8s/pull/3434/commits/ef8e44ae44fc131d3bab5decf3fd9bbce4a3bc33#diff-ba562f0a244dfbd95791e3d82a24c65b87721dbf89ea2144c4c4b30d2b2cbae1R68 Added

Reply:
@sarahalsmiller I think Andrew is saying we should add a second kubebuilder validation for this `port` field to set the min/max value for it

Reply:
Ah my bad! I was in a hurry and misread. https://github.com/hashicorp/consul/pull/20149 <- to take care of that kubetag

Reply:
https://github.com/hashicorp/consul-k8s/pull/3434/files#diff-ba562f0a244dfbd95791e3d82a24c65b87721dbf89ea2144c4c4b30d2b2cbae1R66 added